yarn-project/foundation/src/config/index.ts

```diff
@@ -0,0 +1,111 @@
+export interface ConfigMapping {
+  env?: string;
+  parseEnv?: (val: string) => any;
+  default?: any;
+  printDefault?: (val: any) => string;
+  description: string;
+  isBoolean?: boolean;
+}
+
+export function isBooleanConfigValue<T>(obj: T, key: keyof T): boolean {
+  return typeof obj[key] === 'boolean';
+}
+
+export type ConfigMappingsType<T> = Record<keyof T, ConfigMapping>;
+
+export function getConfigFromMappings<T>(configMappings: ConfigMappingsType<T>): T {
+  const config = {} as T;
+
+  for (const key in configMappings) {
+    if (configMappings[key]) {
+      const { env, parseEnv, default: def } = configMappings[key];
+      // Special case for L1 contract addresses which is an object of config values
+      if (key === 'l1Contracts' && def) {
+        (config as any)[key] = getConfigFromMappings(def);
+      } else {
+        const val = env ? process.env[env] : undefined;
+        if (val !== undefined) {
+          (config as any)[key] = parseEnv ? parseEnv(val) : val;
+        } else if (def !== undefined) {
+          (config as any)[key] = def;
+        }
+      }
+    }
+  }
+
+  return config;
+}
+
+/**
+ * Filters out a service's config mappings to exclude certain keys.
+ * @param configMappings - The service's config mappings
+ * @param keysToFilter - The keys to filter out
+ * @returns The filtered config mappings
+ */
+export function filterConfigMappings<T, K extends keyof T>(
+  configMappings: ConfigMappingsType<T>,
+  keysToFilter: K[],
+): ConfigMappingsType<Omit<T, K>> {
+  return Object.fromEntries(
+    Object.entries(configMappings).filter(([key]) => !keysToFilter.includes(key as K)),
+  ) as ConfigMappingsType<Omit<T, K>>;
+}
+
+/**
+ * Generates parseEnv and default values for a numerical config value.
+ * @param defaultVal - The default numerical value to use if the environment variable is not set or is invalid
+ * @returns Object with parseEnv and default values for a numerical config value
+ */
+export function numberConfigHelper(defaultVal: number): Partial<ConfigMapping> {
+  return {
+    parseEnv: (val: string) => safeParseNumber(val, defaultVal),
+    default: defaultVal,
+  };
+}
```

Comment:
This is great! In a not-too-distant future we should add config helpers for Ethereum address, Aztec address and `Fr` values too :D

Reply:
considered it, PR was getting too big already though 😅